utils/useful.py

```diff
@@ -237,80 +256,93 @@ def created_at(self) -> datetime.datetime:
             raise commands.BotMissingPermissions(["embed_links"])
-        send_dict = {'tts': False, 'file': None, 'files': None, 
+        send_dict = {'tts': False, 'file': None, 'files': None,
                      'delete_after': None, 'nonce': None}
         for x, v in kwargs.items():
             if x in send_dict:
                 send_dict[x] = v
 
         return await to_send(content, mention_author=mention_author, embed=ori_embed, **send_dict)
 
-    def confirmed(self, message_id: Optional[int] = None) -> Coroutine:
+    def confirmed(self, message_id: Optional[int] = None) -> Awaitable[None]:
         message = self.message if not message_id else self.channel.get_partial_message(message_id)
-        return message.add_reaction("<:checkmark:753619798021373974>")
+        # discord.py adds method aliases to PartialMessage which accept Message as self, wrong typing
+        return message.add_reaction("<:checkmark:753619798021373974>")  # type: ignore[misc]
 
-    async def confirmation(self, content: str, delete_after: Optional[bool] = False, *, to_respond: Optional[discord.Member] = None, **kwargs: Any) -> Optional[bool]:
+    async def confirmation(self, content: str, delete_after: Optional[bool] = False, *,
+                           to_respond: Optional[Union[discord.User, discord.Member]] = None,
+                           **kwargs: Any) -> Optional[bool]:
         from utils.buttons import ConfirmView
         return await ConfirmView(self, to_respond=to_respond, delete_after=delete_after).send(content, **kwargs)
 
-    def breaktyping(self, /, *, limit: Optional[int] = None):
+    def breaktyping(self, /, *, limit: Optional[int] = None) -> BreakableTyping:
         return BreakableTyping(self, limit=limit)
 
 
-async def maybe_method(func: Union[Awaitable, Callable], cls: Optional[Type] = None, *args: Any, **kwargs: Any) -> Any:
+async def maybe_method(func: Union[Awaitable[Any], Callable[..., Any]], cls: Optional[type] = None,
+                       *args: Any, **kwargs: Any) -> Any:
     """Pass the class if func is not a method."""
     if not inspect.ismethod(func):
-        return await maybe_coroutine(func, cls, *args, **kwargs)
-    return await maybe_coroutine(func, *args, **kwargs)
+        return await maybe_coroutine(func, cls, *args, **kwargs)  # type: ignore[no-untyped-call]
+    return await maybe_coroutine(func, *args, **kwargs)  # type: ignore[no-untyped-call]
 
 
 @pages()
-def empty_page_format(_, __, entry: T) -> T:
+def empty_page_format(_: Any, __: Any, entry: T) -> T:
     """This is for Code Block ListPageSource and for help Cog ListPageSource"""
     return entry
 
 
-class ListCall(list):
+class ListCall(List[Any]):
     """Quick data structure for calling every element in the array regardless of awaitable or not"""
-    def append(self, rhs: Awaitable) -> list:
-        return super().append(rhs)
+    def append(self, rhs: Awaitable[Any]) -> None:
+        super().append(rhs)
```

Comment:
why do you remove the return i need it chainable

Reply:
fixed